### ISSUE TITLE
Question about Fig 3 in In-context operator learning with data prompts for differential equation problems

### ISSUE THREAD

**Issue by XiangZheng2002** [LiuYangMage/in-context-operator-networks#2]
Hi, thank you for the awesome work! I have a quick question about reproducing Fig. 3a for ODE2/3.
I followed the instructions and used the script in datagen.sh to generate the ODE2/3 datasets. Then, I used the corresponding script in run_group.sh to train the model. However, the results I obtained were not as expected.
Specifically, for ODE2, I used the following command:
gpu=0
dir=data0511a
testeqns=100
testquests=5
traineqns=1000
CUDA_VISIBLE_DEVICES=$gpu python3 datagen.py --dir $dir --name test --eqns $testeqns --quests $testquests --eqn_types ode_auto_linear1 --seed 103 &&
CUDA_VISIBLE_DEVICES=$gpu python3 datagen.py --dir $dir --name train --eqns $traineqns --eqn_types ode_auto_linear1 --seed 3 &&

gpu=0
k_dim=3
k_mode=itx
problem=group-$k_dim-$k_mode
CUDA_VISIBLE_DEVICES=$gpu python3 run.py --problem $problem --num_heads 8 --num_layers 6 --hidden_dim 256 --train_batch_size 16 --epochs 20 --train_warmup_percent 40  --train_data_dirs './data_generation/data0511a' --k_dim $k_dim --k_mode $k_mode --train_data_globs 'train*ode*linear1*' --test_data_globs 'test*ode*linear1*' --tfboard >out_0511a_${problem}_odes_l1.log
Also, I changed the number of epochs to 100 but it still doesn't work very well.
The prediction I plot is like this:
[reproduce_forward.pdf](https://github.com/user-attachments/files/18103544/reproduce_forward.pdf)
[reproduce_inverse.pdf](https://github.com/user-attachments/files/18103545/reproduce_inverse.pdf)
However, the loss looks good, so I'm very confused.
<img width="769" alt="image" src="https://github.com/user-attachments/assets/93486136-dffd-43c5-acfb-3b0f47bfc54a" />
Thanks so much for your clarification and guidance! Looking forward to your response:)

**Comment by LiuYangMage** [comment]
Hi:
Thank you for your interest in our work.
Since the loss looks good, the training should be good (on the training/testing datasets). I cannot tell where exactly the problem is, but I can come up with some possibilities:
1, You used the wrong files for training/plotting, maybe missmatch between the training and plotting datasets. You may want to check the files loaded for training/plotting, and maybe start from plotting the results for training datasets.
2, You didn't use the trained model when plotting.
3, There is something wrong with your plotting code.

**Comment by XiangZheng2002** [comment]
Hi Professor Yang, Thank you so much for your prompt and detailed reply! I looked into the issue and found it's because I selected the wrong dimension for query during model evaluation. Thanks again for your reproducible and awesome works! It really give us a very valuable insight!

**Comment by LiuYangMage** [comment]
Thank you for your kind words! You are welcome!
